transforms/helpers/parse-helper.js

```diff
@@ -473,8 +474,44 @@ function getClassName(
   type = ""
 ) {
   const varDeclaration = getClosetVariableDeclaration(j, eoCallExpression);
+  const filePathSplit = filePath.startsWith("app/")
+    ? filePath.split("app/")
+    : filePath.split("/app/");
+  let normalizedName;
+
+  // Has app in path
+  if (filePathSplit.length === 2) {
+    const appFilePath = filePathSplit[1];
+    const dirWithoutFile = path.dirname(appFilePath);
+    // Only keep no more then the last two directories
+    // Ignore any that are plural of the superclass name
+    const lastTwoDirsOrLess = dirWithoutFile
+      .split("/")
+      .slice(-2)
+      .filter(
+        dir =>
+          capitalizeFirstLetter(camelCase(dir)) !== pluralize(superClassName)
+      );
+
+    const fileNameWithoutExt = path.basename(appFilePath, "js");
+    const superClassNameNotInPath =
+      superClassName &&
+      capitalizeFirstLetter(camelCase(fileNameWithoutExt)) !== superClassName;
```

Comment:
Yeah, maybe a check if the base class is an Ember primitive?

Reply:
I think we need to figure out what the super class is, _and_ what the "kind" is (e.g. Route, Controller, Component, etc). This can be done during our analysis with Dyfactor pretty easily, we may just need to record both and pass them forward.